src/main/java/org/cbio/gdcpipeline/reader/CnaReader.java

```diff
@@ -80,22 +82,39 @@ private void readFile(File cnaFile, ExecutionContext executionContext) throws Ex
         List<String> headerList = new ArrayList<>(Arrays.asList(header.split("\t")));
         headerList.replaceAll(String::toUpperCase);
 
+        // Keep these in order
         int geneIdIndex = headerList.indexOf("GENE ID");
         if (geneIdIndex != -1) {
             headerList.remove(geneIdIndex);
         }
+        int geneSymbolIndex = headerList.indexOf("GENE SYMBOL");
+        if (geneSymbolIndex != -1) {
+            headerList.remove(geneSymbolIndex);
+        }
+        headerList.add(geneSymbolIndex, "Hugo_Symbol");
         int cytobandIndex = headerList.indexOf("CYTOBAND");
         if (cytobandIndex != -1) {
             headerList.remove(cytobandIndex);
         }
 
+        List<Integer> indicesToSkip = new ArrayList<>();
+
         // Replace case ids with sample ids
         for (int i = 0; i < headerList.size(); i++) {
             if (gdcAliquotIdToSampleId.containsKey(headerList.get(i))) {
-                headerList.set(i, gdcAliquotIdToSampleId.get(headerList.get(i)));
+                String sid = gdcAliquotIdToSampleId.get(headerList.get(i));
+                if (samplesSeen.contains(sid)) {
+                    indicesToSkip.add(0,i);
```

Comment:
I suggest skipping the columns in the headerlist here vs. down below. I believe removing the columns the way you are down below would end up removing the wrong columns because the indices aren't being adjusted as you continue removing col's.

Reply:
No it's correct - I add the columns in reverse order so the indexes don't need adjusting

Reply:
oh thanks, I missed that. Then it should be fine